partiql-parser/src/partiql.pest

```diff
@@ -2,18 +2,102 @@ WHITESPACE = _{ " " | "\t" | "\x0B" | "\x0C" | "\r" | "\n" }
 
 COMMENT = _{ LineComment | BlockComment }
 
-BlockComment = { "/*" ~ (BlockComment | (!"*/" ~ ANY))* ~ "*/" }
-LineComment = { "--" ~ (!"\n" ~ ANY )* ~ "\n" }
+BlockComment    = _{ "/*" ~ (BlockComment | (!"*/" ~ ANY))* ~ "*/" }
+LineComment     = _{ "--" ~ (!"\n" ~ ANY )* ~ "\n" }
 
 // TODO implement a full grammar, this is a very primitive version to start
 //      working with Pest and its APIs.
 
-// Entry point for full query parsing
-Query = _{ SOI ~ Token+ ~ EOI}
+//
+// Entry point for full query parsing.
+//
+Query = _{ SOI ~ Expression ~ EOI}
+
+//
+// Expressions
+//
+
+Expression = _{
+      SelectExpression
+    | ValueExpression
+}
+
+// TODO finish the SELECT statement
+
+SelectExpression = {
+    Select ~ (From ~ (Where)?)?
+}
+
+Select = _{
+    SelectValue
+}
+
+SelectValue = {
+    Select_ ~ Value_ ~ ValueExpression
+}
+
+// TODO joins
+
+From = {
+    From_ ~ ValueExpression ~ (As_)? ~ (Identifier)?
+}
+
+Where = {
+    Where_ ~ ValueExpression
+}
+
+// TODO all the operators / paths
+
+ValueExpression = {
+      Term
+    | LeftParen_ ~ Expression ~ RightParen_
+}
 
-// Entry point for query "scanning"
+// TODO fill in all the atoms
+
+Term = _{
+      NullLiteral
+    | BooleanLiteral
+    | String
+    | Number
+    | Identifier
+    | ArrayConstructor
+    | BagConstructor
+    | TupleConstructor
+}
+
+ArrayConstructor = {
+    LeftBracket_ ~ Values? ~ RightBracket_
+}
+
+BagConstructor = {
+    LeftDoubleAngle_ ~ Values? ~ RightDoubleAngle_
+}
+
+Values = _{
+    (ValueExpression ~ (Comma_ ~ ValueExpression)* ~ Comma_?)
+}
+
+TupleConstructor = {
+    LeftCurly_ ~ TupleField ~ (Comma_ ~ TupleField)* ~ Comma_? ~ RightCurly_
```

Comment:
Trailing commas here too... It probably was intentional, but still worth checking.